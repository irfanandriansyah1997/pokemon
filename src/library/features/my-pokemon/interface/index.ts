export * from './my-pokemon-hooks.interface';
export * from './register-dialog.interface';
