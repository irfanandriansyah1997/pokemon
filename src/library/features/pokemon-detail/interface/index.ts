export * from './pokemon-item.interface';
