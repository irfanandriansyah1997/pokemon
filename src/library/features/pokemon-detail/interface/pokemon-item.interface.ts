/**
 * Pokemon Item Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonItem {
  id: number;
  image: string;
  nickname?: string;
}
