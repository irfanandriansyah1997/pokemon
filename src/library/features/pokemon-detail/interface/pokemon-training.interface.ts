/**
 * Pokemon Training Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.31
 */
export interface IPokemonTrainingProps {
  title: string;
  value: string;
}
