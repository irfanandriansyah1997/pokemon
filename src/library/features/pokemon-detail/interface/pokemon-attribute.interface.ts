import { IPokemonSpecies } from '@/library/interface/pokemon';

/**
 * Pokemon Attribute Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.31
 */
export interface IPokemonAttribute {
  key: string;
  value: string | string[];
}

/**
 * Pokemon Attribute Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.31
 */
export interface IPokemonAttributeParameter {
  height: number;
  pokeSpecies?: IPokemonSpecies;
  weight: number;
}
