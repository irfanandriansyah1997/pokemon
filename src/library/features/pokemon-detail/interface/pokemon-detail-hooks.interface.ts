import { IPokemon, IPokemonSpecies } from '@/library/interface/pokemon';

import { IPokemonDetailReducer } from './pokemon-detail-reducer.interface';

/**
 * Pokemon Detail Hooks
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonDetailHooks {
  action: IPokemonDetailHooksAction;
  state: IPokemonDetailReducer;
}

/**
 * Pokemon Detail Hooks Action Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonDetailHooksAction {
  setPokemon(param: IPokemon | undefined): void;
  setPokemonSpecies(param: IPokemonSpecies): void;
  toggleLoading(param: boolean): void;
}
