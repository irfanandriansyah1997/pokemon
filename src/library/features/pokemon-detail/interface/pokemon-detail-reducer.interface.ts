import { GenActionMap, NullAble } from '@/library/interface/general';
import { IBaseQueryError } from '@/library/interface/gql';
import { IPokemon, IPokemonSpecies } from '@/library/interface/pokemon';

/**
 * Pokemon Detail Reducer
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonDetailReducer {
  error?: IBaseQueryError;
  isLoadingRest: boolean;
  pokemon?: IPokemon;
  selection: number;
}

/**
 * Pokemon Detail Enum Reducers
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export enum IPokemonDetailTypesEnum {
  setLoading = `SET_LOADING`,
  setPokemonState = `SET_POKEMON`,
  setSelection = `SET_SELECTION`,
  setPokemonSpecies = `SET_POKEMON_SPECIES`
}

/**
 * Pokemon Detail Payload Action Reducers
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
type IPokemonDetailPayload = {
  [IPokemonDetailTypesEnum.setLoading]: boolean;
  [IPokemonDetailTypesEnum.setPokemonState]: NullAble<IPokemon>;
  [IPokemonDetailTypesEnum.setSelection]: number;
  [IPokemonDetailTypesEnum.setPokemonSpecies]: IPokemonSpecies;
};

export type IPokemonDetailActionType = GenActionMap<IPokemonDetailPayload>[keyof GenActionMap<IPokemonDetailPayload>];
