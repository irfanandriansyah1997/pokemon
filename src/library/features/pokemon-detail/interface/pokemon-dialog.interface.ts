import { IEventComponent } from '@/library/interface/general';
import { IPokemon, IPokemonSprites } from '@/library/interface/pokemon';

export type IEventOnClose = IEventComponent<'on-close'>;

export type IEventOnRelease = IEventComponent<'on-release'>;

export type IPokemonDialogEvent = (
  event: IEventOnClose | IEventOnRelease
) => void;

/**
 * Pokemon Dialog Props
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.30
 */
export interface IPokemonDialogProps {
  on: IPokemonDialogEvent;
  pokemon?: IPokemon;
  showDialog: boolean;
}

/**
 * Pokemon Dialog Backdrop Props
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.31
 */
export interface IPokemonDialogBackdropProps {
  color?: string;
  show?: boolean;
}

/**
 * Pokemon Top Section Props
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.31
 */
export interface IPokemonTopSectionProps {
  genus: string;
  id: string;
  name: string;
  showImage?: boolean;
  showWrapper?: boolean;
  sprite?: IPokemonSprites;
}
