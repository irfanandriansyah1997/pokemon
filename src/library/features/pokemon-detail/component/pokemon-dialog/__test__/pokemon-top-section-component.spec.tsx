import { mount } from 'enzyme';

import PokemonTopSection from '@/library/features/pokemon-detail/component/pokemon-dialog/section/pokemon-top-section.component';
import { PokemonCloseDialog } from '@/library/styles/pokemon.styles';

import { IPokemonTopSectionProps } from '../../../interface';

const DEFAULT_PROPS: IPokemonTopSectionProps = {
  genus: `Seed PokÃ©mon`,
  id: `1`,
  name: `bulbasaur`,
  on: jest.fn(),
  showImage: true,
  showWrapper: true,
  sprite: {
    back_default: `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/1.png`,
    front_default: `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png`
  }
};

describe(`Testing Pokemon Top Section Component`, () => {
  describe(`Testing Snapshot Test`, () => {
    it(`Testing With Set showImage & showWrapper equal true`, () => {
      const onSpy = jest.fn();
      const ui = mount(<PokemonTopSection {...DEFAULT_PROPS} on={onSpy} />);

      expect(ui.html()).toMatchSnapshot();
    });

    it(`Testing With Set showImage equal true`, () => {
      const onSpy = jest.fn();
      const ui = mount(
        <PokemonTopSection {...DEFAULT_PROPS} showWrapper={false} on={onSpy} />
      );

      expect(ui.html()).toMatchSnapshot();
    });

    it(`Testing With Set showWrapper equal true`, () => {
      const onSpy = jest.fn();
      const ui = mount(
        <PokemonTopSection {...DEFAULT_PROPS} showImage={false} on={onSpy} />
      );

      expect(ui.html()).toMatchSnapshot();
    });

    it(`Testing Without sprites`, () => {
      const onSpy = jest.fn();
      const ui = mount(
        <PokemonTopSection {...DEFAULT_PROPS} sprite={undefined} on={onSpy} />
      );

      expect(ui.html()).toMatchSnapshot();
    });
  });

  it(`Testing Simulate Close Dialog`, () => {
    const onSpy = jest.fn((param) => param);
    const ui = mount(
      <PokemonTopSection {...DEFAULT_PROPS} sprite={undefined} on={onSpy} />
    );

    ui.find(PokemonCloseDialog).at(0).simulate(`click`);
    expect(onSpy).toHaveBeenCalledTimes(1);
    expect(onSpy.mock.results[0].value).toStrictEqual({ event: `on-close` });
  });
});
