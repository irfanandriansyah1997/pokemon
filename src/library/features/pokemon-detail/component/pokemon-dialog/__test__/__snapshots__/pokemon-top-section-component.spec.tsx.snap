// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Pokemon Top Section Component Testing Snapshot Test Testing With Set showImage & showWrapper equal true 1`] = `"<div class=\\"css-1way8yj\\"><section class=\\"css-8wlmm2\\"><p class=\\"close css-1r2ireq\\" color=\\"white\\" font-size=\\"normal\\" font-weight=\\"400\\">×</p></section><p color=\\"white\\" font-size=\\"normal\\" font-weight=\\"500\\" class=\\"css-15lnyw7\\">#001</p><p color=\\"white\\" font-size=\\"title\\" font-weight=\\"600\\" class=\\"css-1a6ud9a\\">bulbasaur</p><p class=\\"genus css-5ey957\\" color=\\"white\\" font-size=\\"text\\">Seed Pokémon</p><div class=\\"css-fa00fm\\"><div style=\\"transform: translateX(0%);\\"><div class=\\"css-lvk0ho\\"><img loading=\\"lazy\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png\\" alt=\\"Pokemon\\"></div><div class=\\"css-lvk0ho\\"><img loading=\\"lazy\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/1.png\\" alt=\\"Pokemon\\"></div></div><div class=\\"css-1hulyf\\"></div><div class=\\"css-rm35ms\\"></div><div class=\\"css-qfe1qw\\"><div></div><div></div></div></div></div>"`;

exports[`Testing Pokemon Top Section Component Testing Snapshot Test Testing With Set showImage equal true 1`] = `"<div class=\\"css-73d4k3\\"><section class=\\"css-8wlmm2\\"><p class=\\"close css-5jbmio\\" color=\\"secondary\\" font-size=\\"normal\\" font-weight=\\"400\\">×</p></section><p color=\\"secondary\\" font-size=\\"normal\\" font-weight=\\"500\\" class=\\"css-16tuhca\\">#001</p><p color=\\"secondary\\" font-size=\\"title\\" font-weight=\\"600\\" class=\\"css-jbfcco\\">bulbasaur</p><p class=\\"genus css-1brd6hq\\" color=\\"secondary\\" font-size=\\"text\\">Seed Pokémon</p><div class=\\"css-fa00fm\\"><div style=\\"transform: translateX(0%);\\"><div class=\\"css-lvk0ho\\"><img loading=\\"lazy\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png\\" alt=\\"Pokemon\\"></div><div class=\\"css-lvk0ho\\"><img loading=\\"lazy\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/1.png\\" alt=\\"Pokemon\\"></div></div><div class=\\"css-1hulyf\\"></div><div class=\\"css-rm35ms\\"></div><div class=\\"css-qfe1qw\\"><div></div><div></div></div></div></div>"`;

exports[`Testing Pokemon Top Section Component Testing Snapshot Test Testing With Set showWrapper equal true 1`] = `"<div class=\\"css-1way8yj\\"><section class=\\"css-8wlmm2\\"><p class=\\"close css-1r2ireq\\" color=\\"white\\" font-size=\\"normal\\" font-weight=\\"400\\">×</p></section><p color=\\"white\\" font-size=\\"normal\\" font-weight=\\"500\\" class=\\"css-15lnyw7\\">#001</p><p color=\\"white\\" font-size=\\"title\\" font-weight=\\"600\\" class=\\"css-1a6ud9a\\">bulbasaur</p><p class=\\"genus css-5ey957\\" color=\\"white\\" font-size=\\"text\\">Seed Pokémon</p><div class=\\"css-9eorj8\\"><div style=\\"transform: translateX(0%);\\"><div class=\\"css-lvk0ho\\"><img loading=\\"lazy\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png\\" alt=\\"Pokemon\\"></div><div class=\\"css-lvk0ho\\"><img loading=\\"lazy\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/1.png\\" alt=\\"Pokemon\\"></div></div><div class=\\"css-1hulyf\\"></div><div class=\\"css-rm35ms\\"></div><div class=\\"css-qfe1qw\\"><div></div><div></div></div></div></div>"`;

exports[`Testing Pokemon Top Section Component Testing Snapshot Test Testing Without sprites 1`] = `"<div class=\\"css-1way8yj\\"><section class=\\"css-8wlmm2\\"><p class=\\"close css-1r2ireq\\" color=\\"white\\" font-size=\\"normal\\" font-weight=\\"400\\">×</p></section><p color=\\"white\\" font-size=\\"normal\\" font-weight=\\"500\\" class=\\"css-15lnyw7\\">#001</p><p color=\\"white\\" font-size=\\"title\\" font-weight=\\"600\\" class=\\"css-1a6ud9a\\">bulbasaur</p><p class=\\"genus css-5ey957\\" color=\\"white\\" font-size=\\"text\\">Seed Pokémon</p></div>"`;
